provider/datagen/src/cldr/transform/calendar/japanese.rs

```diff
@@ -24,39 +22,14 @@ const JAPANESE_FILE: &str = include_str!("./snapshot-japanese@1.json");
 /// Common code for a data provider reading from CLDR JSON dates files.
 #[derive(Debug)]
 pub struct JapaneseErasProvider {
-    era_names_path: PathBuf,
-    era_dates_path: PathBuf,
+    source: SourceData,
 }
 
-impl JapaneseErasProvider {
-    /// Constructs an instance from paths to source data.
-    pub fn try_new(cldr_paths: &(impl CldrPaths + ?Sized)) -> eyre::Result<Self> {
-        // The era codes depend on the Latin romanizations of the eras, found
-        // in the `en` locale. We load this data to construct era codes but
-        // actual user code only needs to load the data for the locales it cares about.
-        Ok(Self {
-            era_names_path: cldr_paths
-                .cldr_dates_japanese()?
-                .join("main")
-                .join("en")
-                .join("ca-japanese.json"),
-            era_dates_path: cldr_paths
-                .cldr_core()?
-                .join("supplemental")
-                .join("calendarData.json"),
-        })
-    }
-}
-
-impl TryFrom<&crate::DatagenOptions> for JapaneseErasProvider {
-    type Error = eyre::ErrReport;
-    fn try_from(options: &crate::DatagenOptions) -> eyre::Result<Self> {
-        JapaneseErasProvider::try_new(
-            &**options
-                .cldr_paths
-                .as_ref()
-                .ok_or_else(|| eyre::eyre!("JapaneseErasProvider requires cldr_paths"))?,
-        )
+impl From<&SourceData> for JapaneseErasProvider {
+    fn from(source: &SourceData) -> Self {
+        Self {
+            source: source.clone(),
```

Comment:
I made these all by reference so that we can move to a shared `SourceData` object in the future. This would be the serde cache you were thinking about. I didn't go ahead with sharing this yet though as it would add lifetimes to all transformers.